docs/build/getting-started/testing.md

```diff
@@ -0,0 +1,108 @@
+
+```bash
+mkdir test-cadence
+cd test-cadence
+flow init
+```
+
+## Write a simple smart contract
+
+In your code editor, create a new file called `calculator.cdc` and add the following code:
+
+```cadence calculator.cdc
+access(all) contract Calculator {
+    access(all) fun add(a: Int, b: Int): Int {
+        return a + b;
+    }
+
+    access(all) fun subtract(a: Int, b: Int): Int {
+        return a - b;
+    }
+}
+```
+
+## Add the smart contract to the config
+
+Run `flow config add contract` and add the created `calculator.cdc` contract to the config with the name as `Calculator`. This command adds your contract name and location under the `contracts` key in `flow.json`.
+
+```
+Enter name: Calculate
+Enter contract file location: ./calculator.cdc
+```
+
+## Write test cases
+
+In the same directory, create a new file called `calculator_test.cdc` and add the following code:
+
+```cadence calculator_test.cdc
+import Test
+import "Calculator" // contract name from the previous step
+
+access(all) let calculator = Calculator()
+
+access(all) fun testAdd() {
+  Test.assertEqual(calculator.add(a: 2, b: 3), 5)
+}
+
+access(all) fun testSubtract() {
+  Test.assertEqual(calculator.subtract(a: 5, b: 3), 2)
+}
+
+```
+
+This code
+- imports the `Calculator` contract from the `calculator.cdc` file (according to `flow.json`); 
+- creates an `calculator` instance of the smart-contract;
+- defines two test functions: `testAdd()` and `testSubtract()`;
+- calls `add()` and `subtract()` methods with different input values  respectively.
+
+## Running the test cases
+
+To run the test cases, use the following command in the terminal:
+
+```bash
+flow test ./calculator_test.cdc
+```
+
+This command uses the Flow CLI to run the test cases and display the output. You should see the following output:
+
+```
+Test results: "./calculator_test.cdc"
+- PASS: testAdd
+- PASS: testSubtract
+```
+
+This output indicates that both test cases ran successfully, and the smart contract is functioning as expected.
+
+## Advanced Testing Techniques
+
+The Cadence testing framework provides various features and techniques for writing comprehensive test cases. Some of these include:
+
+- [**Code Coverage**](https://github.com/m-Peter/flow-code-coverage): You can use the `-cover` flag with the `flow test` command to view code coverage results when running your tests. This allows you to identify areas of your code that are not adequately covered by your test inputs;
+- **Test Fixtures**: Test fixtures are reusable components that help you set up the initial state for your test cases. You can create test fixtures in Cadence by defining resource types and using them in your test functions;
+- [**Assertions**](../../cadence/testing-framework.mdx#assertions): The testing framework provides built-in assertion functions, such as `assertEqual`, `assertTrue`, and `assertFalse`, to help you verify the expected behavior of your smart contracts;
```

Comment:
We do not yet have dedicated assertions for `assertTrue` and `assertFalse`. The way to do this is with:
```cadence
let equal = 1 == 1
Test.assert(equal)
Test.assert(!equal)
Test.assert(equal == false)
```
But we have many other assertions worth mentioning, `Test.beNil` / `Test.beEmpty` / `Test.contain` / `Test.beLessThan`, to name a few.